### PR TITLE
Implement proposer slashing

### PR DESCRIPTION
**Pull request by cemozerr**
## PR Description
Implements fraudulent block proposer behavior detection and Proposer Slashing creation. As a result, we are able to correctly slash fraudulent block proposers. Also, adds in configurable naughtiness, which allows us to make a certain set of our validators to act fraudulently.

### CODE REVIEW

File: validator/coordinator/src/main/java/tech/pegasys/artemis/validator/coordinator/ValidatorCoordinator.java [hunk 1]
```diff
@@ -251,15 +303,35 @@ private void createBlockIfNecessary(BeaconStateWithCache state, BeaconBlock oldB
...
         stateTransition.initiate(state, newBlock);
         Bytes32 stateRoot = state.hash_tree_root();
         newBlock.setState_root(stateRoot);
         BLSSignature blockSignature = getBlockSignature(state, newBlock, proposer);
         newBlock.setSignature(blockSignature);
         validatorBlock = newBlock;
+        // TODO: Uncomment this comment block to test proposer slashing
+        // Creates duplicate blocks for our proposer to get it slashed.
+        /*
+        BeaconBlock newestBlock = createInitialBlock(otherState, oldBlock);
```
Comment (jrhea) [inline validator/coordinator/src/main/java/tech/pegasys/artemis/validator/coordinator/ValidatorCoordinator.java]: this is clever.  maybe we should have a naughty flag

**Review by jrhea — APPROVED** [review]
lgtm

File: ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/StateProcessor.java [hunk 2]
```diff
@@ -194,6 +195,10 @@ public void onNewSlot(Date date) throws StateTransitionException, InterruptedExc
         new HeadStateEvent(
             BeaconStateWithCache.deepCopy((BeaconStateWithCache) headState), headBlock));
     recordData(date);
+    System.out.println(
```
Comment (dangerousfood) [inline ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/StateProcessor.java]: Why use `System.out.println` over `LOG.log`?

**Review by dangerousfood — APPROVED** [review]
lgtm

File: ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/StateProcessor.java (hunk 2 above)
Comment (cemozerr) [reply ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/StateProcessor.java]: I gotta clean that. Good catch 🤜

**Pull request merged**